src/cbsodatav4_kcvanderlinden/cbsodatav4.py

```diff
@@ -2,142 +2,188 @@
+    """
+    Return a Pandas DataFrame containing the data from the CBS OData API.
+
+    Parameters
+    ----------
+    tableID : str
+        The ID of the table to retrieve data from.
+    name : str, optional
+        The name of the specific table to retrieve, by default None.
+        If not specified, the entire dataset will be retrieved.
+    limit : int, optional
+        The maximum number of rows to retrieve, by default None.
+        If not specified, all rows will be retrieved.
+    dataFilter : str, optional
+        A filter to apply to the data, by default None.
+        If not specified, no filter will be applied.
+    save_csv_path : str, optional
+        if not None, save the csv file to the specified directory. Path must incude the filename.
+
+    Returns
+    -------
+    Pandas DataFrame
+        A DataFrame containing the retrieved data.
+    """
+    
+    if name == None:
+        df = fullDataset(tableID, limit, dataFilter)
+    else:
+        df = specificTable(tableID, name, limit, dataFilter)
+    # if cache is true, check if cache folder exists, otherwise create it
+    if save_csv_path is not None:
+        df.to_csv(f"{save_csv_path}.csv", index=False, single_file=True)
+        sys.stdout.write(f'\nwritten table to csv')
+    df = df.compute()
+    delete_json_files()
+    return df
+
 def cbsConnect(target_url:str):
     '''
     This function connects to the CBS API and returns a list of dictionaries.
-    target_url: str, the URL is the url that contains the data you want to retrieve
-    returns: list
+    
+    Parameters
+    ----------
+    target_url : str
+        The URL of the data you want to retrieve.
+    Returns
+    -------
+    list
+        A list of dictionaries containing the retrieved data.
+    
+    Examples
+    --------
+    >>> cbsConnect('https://datasets.cbs.nl/odata/v1/CBS/$metadata')
     '''
     session = requests.Session()
     retries = Retry(total=5, backoff_factor=1, status_forcelist=[ 500, 502, 503, 504 ])
     session.mount('https://', HTTPAdapter(max_retries=retries))
     retry_count = 0
-    while retry_count < 5:
+    time_out_time = 0
+    while True:
         try:
             response = session.get(target_url)
             if response.status_code == 200:
-                return response.json()['value']
+                data = response.json()  # Load the JSON data directly from the response object
+                return data
             else:
                 raise requests.exceptions.RequestException(f"HTTP {response.status_code}")
         except requests.exceptions.ChunkedEncodingError:
-            print(f"ChunkedEncodingError occurred. Retrying...")
+            sys.stdout.write(f"ChunkedEncodingError occurred (data load incomplete). Retrying...")
             retry_count += 1
+            time_out_time += random.randrange(1, 10)/10 # dynamic time-out after unstable connection with host.
+            sleep(time_out_time)
             if retry_count == 5:
```

Comment:
Wil je dit nog mee kunnen geven, met 5 als default. Of is 5 eigenlijk altijd wel het logische getal om aan te houden (heb zelf geen idee hoor, meer een vraag ;-))?